tests/foreman/api/test_smartproxy.py

```diff
@@ -14,14 +14,224 @@
 
 @Upstream: No
 """
+from fauxfactory import gen_string
 from nailgun import entities
+from requests import HTTPError
+
+from robottelo.api.utils import one_to_many_names
+from robottelo.cleanup import capsule_cleanup
 from robottelo.config import settings
+from robottelo.datafactory import valid_data_list
 from robottelo.decorators import (
+    bz_bug_is_open,
+    run_only_on,
     skip_if_bug_open,
+    skip_if_not_set,
     tier1,
+    tier2,
+)
+from robottelo.helpers import (
+    default_url_on_new_port,
+    get_available_capsule_port,
 )
 from robottelo.test import APITestCase
-from robottelo.api.utils import one_to_many_names
+
+
+class CapsuleTestCase(APITestCase):
+    """Tests for Smart Proxy (Capsule) entity."""
+
+    @skip_if_not_set('fake_capsules')
+    @run_only_on('sat')
+    @tier1
+    def test_negative_create_with_url(self):
+        """Proxy creation with random URL
+
+        @id: e48a6260-97e0-4234-a69c-77bbbcde85d6
+
+        @Assert: Proxy is not created
+        """
+        # Create a random proxy
+        with self.assertRaises(HTTPError) as context:
+            entities.SmartProxy(
+                url='http://{0}:{1}'.format(
```

Comment:
can't we use `gen_url` or even do not provide any value and necessary url will be generated by nailgun